tests/libgit/test_libgit_spec.lua

```diff
@@ -11,42 +31,160 @@ describe("Libgit2 core methods Test", function()
   end)
 
   before_each(function()
-    MWINDOW_SIZE = lib.luagit2_SET_MWINDOW_SIZE(1234)
-    MWINDOW_MAPPED_LIMIT = lib.luagit2_SET_MWINDOW_MAPPED_LIMIT(1234)
+    lib.luagit2_SET_MWINDOW_SIZE(MWINDOW_SIZE)
+    lib.luagit2_SET_MWINDOW_MAPPED_LIMIT(MWINDOW_MAPPED_LIMIT)
+    lib.luagit2_SET_SEARCH_PATH(1,Search_Paths.System_lvl)
+    lib.luagit2_SET_SEARCH_PATH(2,Search_Paths.Global_lvl)
+    lib.luagit2_SET_SEARCH_PATH(3,Search_Paths.XDG_lvl)
+    lib.luagit2_SET_SEARCH_PATH(4,Search_Paths.Program_data_lvl)
+    lib.luagit2_SET_CACHE_MAX_SIZE(Cache_Max_Size)
+    lib.luagit2_SET_TEMPLATE_PATH(Template_path)
+    lib.luagit2_SET_WINDOWS_SHAREMODE(Windows_Share_Mode)
   end)
```

Comment:
These don't need need to run before each. Maybe they should go into setup or if it should be inside the test they belong to :)

Reply:
Ok I will do so.